corehq/motech/repeaters/repeater_generators.py

```diff
@@ -548,17 +609,21 @@ def _get_target_cases(self, repeat_record, configs, couch_user):
         main_config = configs[0]
         registry_slug = main_config.registry_slug
         helper = DataRegistryHelper(main_config.intent_case.domain, registry_slug=registry_slug)
-        return [
+        return list(filter(None, [
             self._get_case(helper, repeat_record, config, couch_user)
             for config in configs
-        ]
+        ]))
 
     def _get_case(self, registry_helper, repeat_record, config, couch_user):
         try:
             case = registry_helper.get_case(config.case_id, config.case_type, couch_user, repeat_record.repeater)
         except RegistryAccessException:
             raise DataRegistryCaseUpdateError("User does not have permission to access the registry")
+        except CaseTypeMismatch:
+            raise DataRegistryCaseUpdateError(f"Target case not found: {config.case_id}")
         except CaseNotFound:
+            if config.create_case:
+                return
```

Comment:
What about the reverse condition?  If the intent case says "create a new case", but one already exists with that ID.

Reply:
Another good questions that I will have to follow up on.

Reply:
https://github.com/dimagi/commcare-hq/pull/30729